nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/filters/NoVersionFilter.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * (C) Copyright 2006-2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ * Contributors:
+ *    Andrei Nechaev
+ *
+ */
+package org.nuxeo.ai.pipes.filters;
+
+import org.nuxeo.ecm.core.api.DocumentModel;
+
+public class NoVersionFilter implements Filter.DocumentFilter {
+
+    @Override
+    public boolean test(DocumentModel doc) {
+        return !doc.isVersion();
```

Comment:
By anticipation of the next bug, I would add a check actually on isProxy too: proxies are published document that cannot be updated either (as far as i remember)

Reply:
we already have a filter for that kind https://github.com/nuxeo/nuxeo-ai/blob/master/nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/filters/NotSystemOrProxyFilter.java